mutiny-core/src/node.rs

```diff
@@ -569,6 +569,68 @@ impl Node {
         amount_sat: Option<u64>,
         description: String,
         route_hints: Option<Vec<PhantomRouteHints>>,
+    ) -> Result<Invoice, MutinyError> {
+        // the amount to create for the invoice whether or not there is an lsp
+        let (amount_sat, lsp_fee_sat) = if let Some(lsp) = self.lsp_client.clone() {
+            // LSP requires an amount
+            let amount_sat = amount_sat
+                .ok_or(MutinyError::BadAmountError)
+                .and_then(|amount| {
+                    if amount == 0 {
+                        Err(MutinyError::BadAmountError)
+                    } else {
+                        Ok(amount)
+                    }
+                })?;
+
+            // check the fee from the LSP
+            let lsp_fee = lsp
+                .get_lsp_fee_sat(FeeRequest {
+                    pubkey: self.pubkey.to_string(),
+                    amount_msat: amount_sat * 1000,
+                })
+                .await?;
+            let amount_minus_fee = amount_sat
+                .checked_sub(lsp_fee)
+                .ok_or(MutinyError::BadAmountError)?;
+            (Some(amount_minus_fee), Some(lsp_fee))
+        } else {
+            (amount_sat, None)
+        };
+
+        let invoice = self
+            .create_internal_invoice(amount_sat, lsp_fee_sat, description, route_hints)
+            .await?;
+
+        if let Some(lsp) = self.lsp_client.clone() {
+            self.connect_peer(PubkeyConnectionInfo::new(&lsp.connection_string)?, None)
+                .await?;
```

Comment:
should check that we aren't connected already, I think we do this at the nodemanager level, not the node level

Reply:
Not, this calls `connect_peer_if_necessary` so it does that check